### PR TITLE
Improved function calls on x86

### PR DESCRIPTION
**Pull request by elicn**
Tons of changes, mostly to x86 architecture; a partial change list:
- Function call arguments are better recognized
- Functions now show their arguments list and local variables
- Reduced unnecessary dereferences and parenthesis

### CODE REVIEW

**Comment by elicn** [comment]
The new parsing method turns the `parsed` property from an array of strings into a dictionary. Although it offers a cleaner approach, it breaks all architectures that haven't comply yet.

**Comment by wargio** [comment]
i had to move everything under development branch. it broke a lot of things...

File: libdec/arch/x86.js [hunk 1]
```diff
@@ -727,46 +778,80 @@ module.exports = (function() {
...
+
+            return {
+                mnem: mnemonic,
+                opd1: operand1,
+                opd2: operand2
+            };
         },
         context: function() {
+            var JSON = require('libdec/json64');
+            var afvj = JSON.parse(r2cmd('afvj').trim());
```
[the remark this answers is not in the document]
Comment (elicn) [reply libdec/arch/x86.js]: Any idea how to import `r2cmdj` into x86.js?
Perhaps making it available through another module? [currently available only in `r2dec-duk.js`]

[the remark this answers is not in the document]
Comment (wargio) [reply libdec/arch/x86.js]: it was shared in the main scope, so it was possible to use it by default.
please check the https://github.com/wargio/r2dec-js/tree/development branch

**Pull request merged**